geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskRegionAttributesIntegrationTest.java

```diff
@@ -0,0 +1,413 @@
+ */
+public class DiskRegionAttributesIntegrationTest {
+
+  private static final long MAX_OPLOG_SIZE_IN_BYTES = 1024 * 1024 * 1024 * 10L;
+
+  private InternalCache cache;
+  private EvictionAttributes evictionAttributes;
+
+  private File[] diskDirs;
+  private int[] diskDirSizes;
+
+  private String regionName;
+  private String diskStoreName;
+
+  @Rule
+  public ErrorCollector errorCollector = new ErrorCollector();
+
+  @Rule
+  public ExecutorServiceRule executorServiceRule = new ExecutorServiceRule();
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public TestName testName = new TestName();
+
+
+  @Before
+  public void setUp() throws Exception {
+    String uniqueName = getClass().getSimpleName() + "_" + testName.getMethodName();
+    regionName = uniqueName + "_region";
+    diskStoreName = uniqueName + "_diskStore";
+
+    Properties config = new Properties();
+    config.setProperty(MCAST_PORT, "0");
+    config.setProperty(LOCATORS, "");
+
+    cache = (InternalCache) new CacheFactory(config).create();
+
+    diskDirs = new File[4];
+    diskDirs[0] = createDirectory(temporaryFolder.getRoot(), testName.getMethodName() + "1");
+    diskDirs[1] = createDirectory(temporaryFolder.getRoot(), testName.getMethodName() + "2");
+    diskDirs[2] = createDirectory(temporaryFolder.getRoot(), testName.getMethodName() + "3");
+    diskDirs[3] = createDirectory(temporaryFolder.getRoot(), testName.getMethodName() + "4");
+
+    // set default values of disk dir sizes here
+    diskDirSizes = new int[4];
+    diskDirSizes[0] = Integer.MAX_VALUE;
+    diskDirSizes[1] = Integer.MAX_VALUE;
+    diskDirSizes[2] = Integer.MAX_VALUE;
+    diskDirSizes[3] = Integer.MAX_VALUE;
+
+    evictionAttributes = createLRUEntryAttributes(1000, OVERFLOW_TO_DISK);
+
+    DiskStoreImpl.SET_IGNORE_PREALLOCATE = true;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    try {
+      if (cache != null) {
+        cache.close();
+      }
+    } finally {
+      DiskStoreImpl.SET_IGNORE_PREALLOCATE = false;
+    }
+  }
+
+  @Test
+  public void syncPersistentWithAutoCompact() {
+    DiskStoreFactory diskStoreFactory = cache.createDiskStoreFactory();
+    diskStoreFactory.setAutoCompact(true);
+    diskStoreFactory.setDiskDirsAndSizes(diskDirs, diskDirSizes);
+
+    createDiskStoreWithSizeInBytes(diskStoreName, diskStoreFactory, 10737418240l);
+
+    RegionFactory<Object, Object> regionFactory = cache.createRegionFactory(LOCAL);
+    regionFactory.setDataPolicy(DataPolicy.PERSISTENT_REPLICATE);
+    regionFactory.setDiskStoreName(diskStoreName);
+    regionFactory.setDiskSynchronous(true);
+
+    Region region = regionFactory.create(regionName);
+
+    assertThat(region.getAttributes().isDiskSynchronous()).isEqualTo(true);
+    verifyRegionAndDiskStoreAttributes(true, 10737418240l, 0, -1);
```

Comment:
Why the mixed use of `MAX_OPLOG_SIZE_IN_BYTES` and the literal `10737418240l` from one test to the next?

Reply:
I copied that from the old test. Several tests use smaller literals and honestly, I just didn't realize that 10737418240l was equal to MAX_OPLOG_SIZE_IN_BYTES (is it?). I'll check if they're the same and change it.

Reply:
Fixed!

Reply:
Thanks. Wasn't sure it was a holdover from from pre-refactor or intentional for a reason I didn't discern.